src/contract.rs

```diff
@@ -205,14 +254,37 @@ pub fn execute_execute(
     prop.status = Status::Executed;
     PROPOSALS.save(deps.storage, proposal_id.into(), &prop)?;
 
+    let refund_msg = get_proposal_deposit_refund_message(&prop.proposer, &prop.deposit)?;
+
     // dispatch all proposed messages
     Ok(Response::new()
+        .add_messages(refund_msg)
         .add_messages(prop.msgs)
         .add_attribute("action", "execute")
         .add_attribute("sender", info.sender)
         .add_attribute("proposal_id", proposal_id.to_string()))
 }
 
+fn get_proposal_deposit_refund_message(
+    proposer: &Addr,
+    config: &ProposalDeposit,
```

Comment:
Making a comment for later, but feel this variable could be named a bit better... like why not `deposit` or `proposal_deposit`? Not a big deal though... good enough for now.